### PR TITLE
Add TM Firewall rules resource module

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

Add TM Firewall rules resource module

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

deepsec_firewall_rules

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/deepsec_firewall_rules.py [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+        description: Source IP type. Default is "any". Searchable as Choice.
+        type: str
+        choices:
+        - any
+        - masked-ip
+        - range
+        - ip-list
+        - single
+        - multiple
+      source_ipvalue:
```
Comment (NilashishC) [inline plugins/modules/deepsec_firewall_rules.py:103]: Can we move all the `source_*` and `destination_*` keys under a dict with the respective keys as parent?

```
source:
  ipvalue:
  ipmask:
destination:
   ipvalue:
   ipmask:
```

File: plugins/modules/deepsec_firewall_rules.py [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+    - merged
+    - replaced
+    - overridden
+    - gathered
+    - deleted
+
+author: Ansible Security Automation Team (@justjais) <https://github.com/ansible-security>
+"""
+
+EXAMPLES = """
```
Comment (NilashishC) [inline plugins/modules/deepsec_firewall_rules.py:314]: Does this need a RETURN_VALUES key?

File: plugins/modules/deepsec_firewall_rules.py (hunk 1 above)
Comment (justjais) [reply plugins/modules/deepsec_firewall_rules.py:103]: Here for security platforms, we generate module from their REST API and thus replicate the module params from REST doc, coz of which we're not planning on making structural changes to available params.

File: plugins/modules/deepsec_firewall_rules.py (hunk 2 above)
Comment (justjais) [reply plugins/modules/deepsec_firewall_rules.py:314]: Added, thanks

**Comment by NilashishC** [comment]
@justjais +1

**Pull request merged**